statemachine/src/main/java/com/amazonaws/stepfunctions/cloudformation/statemachine/DefinitionProcessor.java

```diff
@@ -129,9 +135,15 @@ private static String fetchS3Definition(final S3Location s3Location, final Amazo
         return definition;
     }
 
-    private static String convertDefinitionObjectToString(final Map<String, Object> definitionObject) {
+    private static String convertDefinitionObjectToString(final ResourceModel model) {
+        final Map<String, Object> definitionObject = model.getDefinition();
         try {
-            return jsonMapper.writerWithDefaultPrettyPrinter().writeValueAsString(definitionObject);
+            String definition = jsonMapper.writerWithDefaultPrettyPrinter().writeValueAsString(definitionObject);
+            // DefinitionSubstitution before validating JSON or YAML format
```

Comment:
Question (non-blocking): do we validate the JSON or YAML format after this comment?

Reply:
Nope, I'll change the comment in my next PR